homeassistant/components/overkiz/climate_entities/somfy_thermostat.py

```diff
@@ -0,0 +1,178 @@
+"""Support for TaHoma Smart Thermostat."""
+from __future__ import annotations
+
+from typing import Any, cast
+
+from pyoverkiz.enums import OverkizState
+
+from homeassistant.components.climate import ClimateEntity
+from homeassistant.components.climate.const import (
+    CURRENT_HVAC_HEAT,
+    CURRENT_HVAC_IDLE,
+    HVAC_MODE_AUTO,
+    HVAC_MODE_HEAT,
+    PRESET_AWAY,
+    PRESET_HOME,
+    PRESET_NONE,
+    SUPPORT_PRESET_MODE,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+
+from ..coordinator import OverkizDataUpdateCoordinator
+from ..entity import OverkizEntity
+
+COMMAND_EXIT_DEROGATION = "exitDerogation"
+COMMAND_REFRESH_STATE = "refreshState"
+COMMAND_SET_DEROGATION = "setDerogation"
+COMMAND_SET_MODE_TEMPERATURE = "setModeTemperature"
+
+CORE_DEROGATED_TARGET_TEMPERATURE_STATE = "core:DerogatedTargetTemperatureState"
+CORE_DEROGATION_ACTIVATION_STATE = "core:DerogationActivationState"
+
+PRESET_FREEZE = "freeze"
+PRESET_NIGHT = "night"
+
+ST_HEATING_MODE_STATE = "somfythermostat:HeatingModeState"
+ST_DEROGATION_HEATING_MODE_STATE = "somfythermostat:DerogationHeatingModeState"
+
+STATE_DEROGATION_FURTHER_NOTICE = "further_notice"
+STATE_DEROGATION_ACTIVE = "active"
+STATE_DEROGATION_INACTIVE = "inactive"
+STATE_PRESET_AT_HOME = "atHomeMode"
+STATE_PRESET_AWAY = "awayMode"
+STATE_PRESET_FREEZE = "freezeMode"
+STATE_PRESET_MANUAL = "manualMode"
+STATE_PRESET_SLEEPING_MODE = "sleepingMode"
+STATE_PRESET_SUDDEN_DROP_MODE = "suddenDropMode"
+
+OVERKIZ_TO_HVAC_MODES = {
+    STATE_DEROGATION_ACTIVE: HVAC_MODE_HEAT,
+    STATE_DEROGATION_INACTIVE: HVAC_MODE_AUTO,
+}
+OVERKIZ_TO_PRESET_MODES = {
+    STATE_PRESET_AT_HOME: PRESET_HOME,
+    STATE_PRESET_AWAY: PRESET_AWAY,
+    STATE_PRESET_FREEZE: PRESET_FREEZE,
+    STATE_PRESET_MANUAL: PRESET_NONE,
+    STATE_PRESET_SLEEPING_MODE: PRESET_NIGHT,
+    STATE_PRESET_SUDDEN_DROP_MODE: PRESET_NONE,
+}
+PRESET_MODES_TO_OVERKIZ = {v: k for k, v in OVERKIZ_TO_PRESET_MODES.items()}
+TARGET_TEMP_TO_OVERKIZ = {
+    PRESET_HOME: "somfythermostat:AtHomeTargetTemperatureState",
+    PRESET_AWAY: "somfythermostat:AwayModeTargetTemperatureState",
+    PRESET_FREEZE: "somfythermostat:FreezeModeTargetTemperatureState",
+    PRESET_NIGHT: "somfythermostat:SleepingModeTargetTemperatureState",
+}
+
+
+class SomfyThermostat(OverkizEntity, ClimateEntity):
+    """Representation of Somfy Smart Thermostat."""
+
+    _attr_temperature_unit = TEMP_CELSIUS
+    _attr_supported_features = SUPPORT_PRESET_MODE | SUPPORT_TARGET_TEMPERATURE
+    _attr_hvac_modes = [HVAC_MODE_AUTO, HVAC_MODE_HEAT]
```

Comment:
```suggestion
    _attr_hvac_modes = [*HVAC_MODES_TO_OVERKIZ]
```

Reply:
For other implementations, we created this one. So that we don't manually need to keep another list of supported hvac_modes / preset_modes.

Reply:
Done.